hera_qm/tests/test_xrfi.py

```diff
@@ -42,31 +42,232 @@
+    uvf = UVFlag(uvdata_miriad_main, mode='flag', use_future_array_shapes=True)
+    yield uvf
+    # clean up when done
+    del uvf
+
+    return
+
+
+@pytest.fixture(scope="function")
+def uvflag_flag_miriad(uvflag_flag_miriad_main):
+    uvm = uvflag_flag_miriad_main.copy()
+    yield uvm
+    # clean up when done
+    del uvm
+
+    return
+
+
+@pytest.fixture(scope="session")
+def uvflag_flag_waterfall_miriad_main(uvdata_miriad_main):
+    uvf = UVFlag(
+        uvdata_miriad_main,
+        mode='flag',
+        waterfall=True,
+        use_future_array_shapes=True
+    )
+    yield uvf
+    # clean up when done
+    del uvf
+
+    return
+
+
+@pytest.fixture(scope="function")
+def uvflag_flag_waterfall_miriad(uvflag_flag_waterfall_miriad_main):
+    uvf = uvflag_flag_waterfall_miriad_main.copy()
+    yield uvf
+    # clean up when done
+    del uvf
+
+    return
+
+
+@pytest.fixture(scope="session")
+def uvcal_calfits_main():
+    uvc = UVCal()
+    uvc.read_calfits(test_c_file, use_future_array_shapes=True)
+    yield uvc
+
+    # clean up when done
+    del uvc
+
+    return
+
+
+@pytest.fixture(scope="function")
+def uvcal_calfits(uvcal_calfits_main):
+    uvc = uvcal_calfits_main.copy()
+    yield uvc
+
+    # clean up when done
+    del uvc
+
+    return
+
+
+@pytest.fixture(scope="session")
+def uvflag_metric_calfits_main(uvcal_calfits_main):
+    uvm = UVFlag(
+        uvcal_calfits_main, history='I made this', use_future_array_shapes=True
+    )
+    yield uvm
+
+    # clean up when done
+    del uvm
+
+    return
+
+
+@pytest.fixture(scope="function")
+def uvflag_metric_calfits(uvflag_metric_calfits_main):
+    uvm = uvflag_metric_calfits_main.copy()
+    yield uvm
+
+    # clean up when done
+    del uvm
+
+    return
+
+
+@pytest.fixture(scope="session")
+def uvflag_flag_calfits_main(uvcal_calfits_main):
+    uvf = UVFlag(uvcal_calfits_main, mode='flag', use_future_array_shapes=True)
+    yield uvf
+
+    # clean up when done
+    del uvf
+
+    return
+
+
+@pytest.fixture(scope="function")
+def uvflag_flag_calfits(uvflag_flag_calfits_main):
+    uvf = uvflag_flag_calfits_main.copy()
+    yield uvf
+
+    # clean up when done
+    del uvf
+
+    return
+
+
+@pytest.fixture(scope="session")
+def uvflag_f_main():
+    uvf = UVFlag(test_f_file, use_future_array_shapes=True)
+    yield uvf
+
+    # clean up when done
+    del uvf
+
+    return
+
+
+@pytest.fixture(scope="function")
+def uvflag_f(uvflag_f_main):
+    uvf = uvflag_f_main.copy()
+    yield uvf
+
+    # clean up when done
+    del uvf
+
+    return
+
+
+def test_uvdata(uvdata_miriad):
+    uv = uvdata_miriad
     xant = uv.get_ants()[0]
     xrfi.flag_xants(uv, xant)
     assert np.all(uv.flag_array[uv.ant_1_array == xant, :, :])
     assert np.all(uv.flag_array[uv.ant_2_array == xant, :, :])
 
 
-def test_uvcal():
-    uvc = UVCal()
-    uvc.read_calfits(test_c_file)
-    uvc.use_future_array_shapes()
+def test_uvcal(uvcal_calfits):
```

Comment:
Mostly out of curiosity, does `uvcal_calfits` have to passed into the function? It's defined as global, right?

Reply:
This is how pytest fixtures work.